internal/docker/build.go

```diff
@@ -67,13 +68,24 @@ func Build(
 		}...)
 	}
 	commonArgs = append(commonArgs, args...)
-	f, err := os.Open(filepath.Join(filepath.Dir(plugin.Path), "Dockerfile"))
-	if err != nil {
-		return nil, err
+	buildArgs := slices.Concat(commonArgs, []string{
+		"-t", imageName,
+		filepath.Dir(plugin.Path),
+	})
+	cmd := exec.CommandContext(ctx, "docker", buildArgs...)
+	// Set file modification times to bust Docker cache for local files
```

Comment:
This is a pretty ugly hack - I'm not able to reproduce it in local builds (only in CI). Verified that this does appear to fix the build and test failures in this PR however.

An alternative might be to create a temporary directory and copy all the Docker context files to it prior to build, but just setting the mtime does the trick too.

Reply:
Would this help us in any way? Should we add this regardless?

https://docs.docker.com/build/ci/github-actions/reproducible-builds/

Reply:
I don't believe that will help in this case. I think the `COPY` command is getting tripped up because the `package.json` and `package-lock.json` files have identical mtime (due to git checkout) and sizes and thinking it can re-use the local cache - https://github.com/moby/buildkit/issues/4817 looks related.

The source epoch would allow our images to be more reproducible across builds which is a good thing but perhaps something we should tweak outside of this change.